src/Evolution/Systems/Cce/BoundaryData.hpp

```diff
@@ -539,4 +539,620 @@ void du_j_worldtube_data(
+      get<::Tags::deriv<gr::Tags::Shift<3, ::Frame::Inertial, DataVector>,
+                        tmpl::size_t<3>, ::Frame::Inertial>>(
+          computation_variables);
+  auto& dt_cartesian_shift =
+      get<::Tags::dt<gr::Tags::Shift<3, ::Frame::Inertial, DataVector>>>(
+          computation_variables);
+
+  cartesian_shift_and_derivatives_from_modes(
+      make_not_null(&cartesian_shift), make_not_null(&d_cartesian_shift),
+      make_not_null(&dt_cartesian_shift),
+      make_not_null(&interpolation_modal_buffer),
+      make_not_null(&interpolation_buffer), make_not_null(&eth_buffer),
+      shift_coefficients, dr_shift_coefficients, dt_shift_coefficients,
+      inverse_cartesian_to_spherical_jacobian, l_max);
+
+  auto& cartesian_lapse =
+      get<gr::Tags::Lapse<DataVector>>(computation_variables);
+  auto& d_cartesian_lapse =
+      get<::Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<3>,
+                        ::Frame::Inertial>>(computation_variables);
+  auto& dt_cartesian_lapse =
+      get<::Tags::dt<gr::Tags::Lapse<DataVector>>>(computation_variables);
+  cartesian_lapse_and_derivatives_from_modes(
+      make_not_null(&cartesian_lapse), make_not_null(&d_cartesian_lapse),
+      make_not_null(&dt_cartesian_lapse),
+      make_not_null(&interpolation_modal_buffer),
+      make_not_null(&interpolation_buffer), make_not_null(&eth_buffer),
+      lapse_coefficients, dr_lapse_coefficients, dt_lapse_coefficients,
+      inverse_cartesian_to_spherical_jacobian, l_max);
+
+  auto& phi = get<GeneralizedHarmonic::Tags::Phi<3, ::Frame::Inertial>>(
+      computation_variables);
+  auto& dt_spacetime_metric = get<
+      ::Tags::dt<gr::Tags::SpacetimeMetric<3, ::Frame::Inertial, DataVector>>>(
+      computation_variables);
+  auto& spacetime_metric =
+      get<gr::Tags::SpacetimeMetric<3, ::Frame::Inertial, DataVector>>(
+          computation_variables);
+  GeneralizedHarmonic::phi(
+      make_not_null(&phi), cartesian_lapse, d_cartesian_lapse, cartesian_shift,
+      d_cartesian_shift, cartesian_spatial_metric, d_cartesian_spatial_metric);
+  gr::time_derivative_of_spacetime_metric(
+      make_not_null(&dt_spacetime_metric), cartesian_lapse, dt_cartesian_lapse,
+      cartesian_shift, dt_cartesian_shift, cartesian_spatial_metric,
+      dt_cartesian_spatial_metric);
+  gr::spacetime_metric(make_not_null(&spacetime_metric), cartesian_lapse,
+                       cartesian_shift, cartesian_spatial_metric);
+
+  auto& dt_worldtube_normal =
+      get<::Tags::dt<Tags::detail::WorldtubeNormal>>(computation_variables);
+  auto& worldtube_normal =
+      get<Tags::detail::WorldtubeNormal>(computation_variables);
+  worldtube_normal_and_derivatives(
+      make_not_null(&worldtube_normal), make_not_null(&dt_worldtube_normal),
+      cos_phi, cos_theta, spacetime_metric, dt_spacetime_metric, sin_phi,
+      sin_theta, inverse_spatial_metric);
+
+  auto& du_null_l = get<::Tags::dt<Tags::detail::NullL>>(computation_variables);
```

Comment:
should this use `Tags::Du` instead of `Tags::dt` ?

Reply:
That might appear more intuitive, but the `Du` tag is only for spin-weighted quantities.